backend/src/Notifo.Domain.Integrations.Abstractions/IntegrationProperty.cs

```diff
@@ -185,6 +185,7 @@ private bool TryGetString(string? input, [MaybeNullWhen(true)] out string error,
 
                     break;
                 case PropertyFormat.Url:
+                    // We only allow http and https to enable the usage of URL field for HttpClient requests.
                     if (!Uri.TryCreate(input, UriKind.Absolute, out var uri) || !((string[])["http", "https"]).Contains(uri.Scheme, StringComparer.OrdinalIgnoreCase))
```

Comment:
Please introduce a static or improve the query. But having an allocation for this is really weird ;)

Reply:
Is there a test for that?

Reply:
What is about mqtt:// and all other URL properties?

Reply:
Hey! Sure, what other URL schemes do you think should be supported, besides http, https, mqtt? 
I didn't add mqtt and other schemes (e.g. smtp) at first since this URL field was specifically supposed to be used for HTTP requests, but sure - we can make it be more universal. Or maybe should we create two PropertyFormats - one for HTTP URLs, one for other types?

Reply:
The URL should support all schemes. If an integration needs a special URL, it could just use patterns I think. You usually also get a runtime error so we should not validate too much.

Reply:
So maybe we can extend the PropertyFormat a bit? Now it's an Enum, but maybe we can make it accept a class instance that implements an abstract `PropertyFormatBase` class. In that class, we'd create format-specific fields, e.g. `AllowedSchemas` for URLs. What do you think?

Reply:
I think this would be overengineered. I would keep it simple. Perhaps just introduce PatternMessage and get rid of the format because it does not work well? Then you can just introduce a few pairs of message and pattern in the backend.

Reply:
Hey, I'm so sorry for not being here for a while, it's because of some independent things, I want to apologize for my absence here.
You're right, that might've been overengineered. But there's one problem with the Pattern approach - what if we need some custom validators? For example for the URLs, we have the `Uri.TryCreate()` method that I guess is the best approach to check if the given URL would be accepted by a `HttpClient`. Technically we could create a regular expression, but using a custom validator seems like a better option here.

What do you think?

Reply:
Just the URL is the worst example actually. Because they are dozens of URL formats, e.g. for databases and so on.

Reply:
If we keep it we should at least rename it to HttpUrl or something like that.

Reply:
Okay, sure. So we can rename it to HttpUrl.
So in the end, do you personally think it's worth keeping the `PropertyFormat`, or should we stick to `PropertyPattern` and just add the custom error message?

Reply:
Lets keep it.

Reply:
Okay, sure! 
And now another question - should we keep this `PropertyFormat` as it is or introduce more customization (as said earlier in this thread) by creating that base `BasePropertyFormat` class and then specific classes for given types, e.g. `UrlFormat` that also has a field `AllowedSchemas` etc.?

Reply:
For me the advantage of the propertyFormat is that we can also use it in the UI and use specialized inputs for example. If we are just using that for validation we would be probably fine with regex. I don't want to make it too complicated.

But if we introduce base classes and stuff like that it makes the UI super complicated.

Reply:
Okay, sure, you're right. Therefore - I'll rename the "Url" `PropertyFormat` to "HttpUrl" and only add http and https as allowed schemas.
Should I also add a generic "Url" `PropertyFormat`? (for input[type="url"] on front-end, so in the end, there will be a separate `Url` and `HttpUrl`)

Reply:
If a URL is not http or https they very like need additional validation, e.g. if it starts with "mongo://" or something that. So HttpUrl is fine.

Reply:
There is still the extra allocation per call.

Reply:
Oh, yeah, sure, I'm so sorry about forgetting to change that😅